flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/connector/elasticsearch/table/IndexGeneratorFactory.java

```diff
@@ -88,6 +89,20 @@ private static IndexGenerator createRuntimeIndexGenerator(
                 index.substring(indexPrefix.length() + dynamicIndexPatternStr.length());
 
         final boolean isDynamicIndexWithFormat = indexHelper.checkIsDynamicIndexWithFormat(index);
+        final boolean isProcTimeDynamicIndex =
+                indexHelper.checkIsProcTimeDynamicIndexWithFormat(index);
+
+        if (isProcTimeDynamicIndex) {
+            final String dateTimeFormat =
+                    indexHelper.extractDateFormat(index, LogicalTypeRoot.TIME_WITHOUT_TIME_ZONE);
```

Comment:
Could you explain why you are using `TIME_WITHOUT_TIME_ZONE` here?

Reply:
I changed to TIMESTAMP_WITHOUT_TIMEZZONE.  As the return type of  now() function is  TIMESTAMP .